Fix <amp-sticky-ad> tests

### Flaky test: should listen to scroll event
Produces error: 
```
DESCRIBE => amp-sticky-ad 1.0 version
  DESCRIBE =>  
    DESCRIBE => with valid child 1.0
      IT => should listen to scroll event
        ✗ AssertionError: expected null to be a function
```

### Console errors
```
        ● should not allow container to be set to transparent
ERROR: 'amp-sticky-ad must have a single amp-ad child'
    The test "amp-sticky-ad 1.0 version   with unvalid child 1.0 should not build when child is not ad" resulted in a call to console.error.
    ⤷ If this is not expected, fix the code that generated the error.
    ⤷ If this is expected, use the following pattern to wrap the test code that generated the error:
        'allowConsoleError(() => { <code that generated the error> });'
      with unvalid child 1.0
        ● should not build when child is not ad
ERROR: 'amp-sticky-ad must have a single amp-ad child'
    The test "amp-sticky-ad 1.0 version   with unvalid child 1.0 should not build when has more than 1 children" resulted in a call to console.error.
    ⤷ If this is not expected, fix the code that generated the error.
    ⤷ If this is expected, use the following pattern to wrap the test code that generated the error:
        'allowConsoleError(() => { <code that generated the error> });'
```

--- a/extensions/amp-sticky-ad/1.0/test/test-amp-sticky-ad.js
+++ b/extensions/amp-sticky-ad/1.0/test/test-amp-sticky-ad.js
@@ -17,6 +17,7 @@
 
 import '../../../amp-ad/0.1/amp-ad';
 import '../amp-sticky-ad';
+import {createElementWithAttributes} from '../../../../src/dom';
 import {macroTask} from '../../../../testing/yield';
 import {poll} from '../../../../testing/iframe';
 
@@ -27,11 +28,12 @@ describes.realWin('amp-sticky-ad 1.0 version', {
   },
   amp: { /* amp spec */
     runtimeOn: false,
-    extensions: ['amp-sticky-ad:1.0'],
+    extensions: ['amp-sticky-ad:1.0', 'amp-ad'],
   },
 }, env => {
   let win;
   let ampStickyAd;
+  let ampAd;
   let impl;
   let addToFixedLayerStub, addToFixedLayerPromise;
   const adUpgradedToCustomElementPromise = Promise.resolve();
@@ -40,7 +42,11 @@ describes.realWin('amp-sticky-ad 1.0 version', {
       win = env.win;
       ampStickyAd = win.document.createElement('amp-sticky-ad');
       ampStickyAd.setAttribute('layout', 'nodisplay');
-      const ampAd = win.document.createElement('amp-ad');
+      ampAd = createElementWithAttributes(win.document, 'amp-ad', {
+        'type': '_ping_',
+        'height': 50,
+        'width': 300,
+      });
       ampStickyAd.appendChild(ampAd);
       win.document.body.appendChild(ampStickyAd);
       ampStickyAd.build();
@@ -51,8 +57,7 @@ describes.realWin('amp-sticky-ad 1.0 version', {
               () => addToFixedLayerPromise);
     });
 
-    // TODO (#14775): fix and unskip this test
-    it.skip('should listen to scroll event', function * () {
+    it('should listen to scroll event', function * () {
       expect(impl.scrollUnlisten_).to.be.null;
       yield macroTask();
       expect(impl.scrollUnlisten_).to.be.a('function');
@@ -254,8 +259,16 @@ describes.realWin('amp-sticky-ad 1.0 version', {
       ampStickyAd = win.document.createElement('amp-sticky-ad');
       ampStickyAd.setAttribute('layout', 'nodisplay');
       ampImg = win.document.createElement('amp-img');
-      ampAd1 = win.document.createElement('amp-ad');
-      ampAd2 = win.document.createElement('amp-ad');
+      ampAd1 = createElementWithAttributes(win.document, 'amp-ad', {
+        'type': '_ping_',
+        'height': 50,
+        'width': 300,
+      });
+      ampAd2 = createElementWithAttributes(win.document, 'amp-ad', {
+        'type': '_ping_',
+        'height': 50,
+        'width': 300,
+      });
       win.document.body.appendChild(ampStickyAd);
     });
 
@@ -302,10 +315,11 @@ describes.realWin('amp-sticky-ad 1.0 with real ad child', {
     win = env.win;
     ampStickyAd = win.document.createElement('amp-sticky-ad');
     ampStickyAd.setAttribute('layout', 'nodisplay');
-    const ampAd = win.document.createElement('amp-ad');
-    ampAd.setAttribute('height', '50');
-    ampAd.setAttribute('width', '200');
-    ampAd.setAttribute('type', '_ping_');
+    const ampAd = createElementWithAttributes(win.document, 'amp-ad', {
+      'type': '_ping_',
+      'height': 50,
+      'width': 200,
+    });
     ampStickyAd.appendChild(ampAd);
     win.document.body.appendChild(ampStickyAd);
     ampStickyAd.build();

--- a/extensions/amp-sticky-ad/1.0/test/test-amp-sticky-ad.js
+++ b/extensions/amp-sticky-ad/1.0/test/test-amp-sticky-ad.js
@@ -58,9 +58,15 @@ describes.realWin('amp-sticky-ad 1.0 version', {
     });
 
     it('should listen to scroll event', function * () {
+      const spy = sandbox.spy(impl, 'removeOnScrollListener_');
       expect(impl.scrollUnlisten_).to.be.null;
       yield macroTask();
-      expect(impl.scrollUnlisten_).to.be.a('function');
+      // Hack to handle possible unexpected page scroll
+      if (impl.scrollUnlisten_) {
+        expect(impl.scrollUnlisten_).to.be.a('function');
+      } else {
+        expect(spy).to.be.calledOnce;
+      }
     });
 
     it('should not build when scrollTop not greater than 1', () => {

